src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -5445,6 +5445,209 @@ static void F2(object? w)
                 Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(14, 26));
         }
 
+        [Fact, WorkItem(35748, "https://github.com/dotnet/roslyn/issues/35748")]
+        public void Directive_Annotations_Enable()
+        {
+            var source = @"
+#nullable enable annotations
+class Program
+{
+    static void M(string? s)
+    {
+        _ = s.ToString();
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem(35748, "https://github.com/dotnet/roslyn/issues/35748")]
+        public void Directive_Annotations_Enable_01()
+        {
+            var source = @"
+#nullable enable annotations
+class Program
+{
+    static void M(string? s)
+    {
+        _ = s.ToString();
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem(35748, "https://github.com/dotnet/roslyn/issues/35748")]
+        public void Directive_Annotations_Enable_02()
+        {
+            var source = @"
+#nullable enable
+#nullable disable warnings
+class Program
+{
+    static void M(string? s)
+    {
+        _ = s.ToString();
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem(35748, "https://github.com/dotnet/roslyn/issues/35748")]
+        public void Directive_Annotations_Enable_03()
+        {
+            var source = @"
+#nullable enable
+#nullable restore warnings
+class Program
+{
+    static void M(string? s)
+    {
+        _ = s.ToString();
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem(35748, "https://github.com/dotnet/roslyn/issues/35748")]
+        public void Directive_Annotations_Enable_04()
+        {
+            var source = @"
+class Program
+{
+    static void M(string? s)
+    {
+        _ = s.ToString();
+    }
+}
+";
+            var comp = CreateCompilation(source, options: WithNonNullTypes(NullableContextOptions.Annotations));
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem(35748, "https://github.com/dotnet/roslyn/issues/35748")]
+        public void Directive_Warnings_Enable_01()
+        {
+            var source = @"
+#nullable enable warnings
+class Program
+{
+    static void M(string? s)
+    {
+        _ = s.ToString();
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                    // (5,25): warning CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' context.
```

Comment:
>The annotation for nullable reference types should only be used in code within a '#nullable' context. [](start = 47, length = 101)

We should probably tweak this message. The user did specify a `#nullable`, but only for warnings context.

Reply:
e.g. `#nullable annotations` context?

Reply:
Maybe "The annotation for nullable reference types should only be used in code within a context that enables nullable annotations (for instance with `#nullable enable` or `#nullable enable annotations`)."

---
In reply to: [294561707](https://github.com/dotnet/roslyn/pull/36464#discussion_r294561707) [](ancestors = 294561707)

Reply:
I'm considering `The annotation for nullable reference types should only be used in code within a '#nullable' annotations context.` and the suggestion to add a nullable directive should be given via code fix.

---
In reply to: [294575863](https://github.com/dotnet/roslyn/pull/36464#discussion_r294575863) [](ancestors = 294575863,294561707)